tests/test_ratmama.py

```diff
@@ -0,0 +1,244 @@
+        monkeypatch.setattr(context_channel_fx, '_words_eol',
+                            ["Incoming Client: Ajdacho - System: H - Platform: XB - "
+                             "O2: OK - Language: Polish (pl-PL)"
+                             ]
+                            )
+
+        # with an invalid nickname
+        monkeypatch.setattr(context_channel_fx._user, '_nickname', "MasterLoon")
+        # have it licked by the handler
+        await RatMama.handle_ratmama_announcement(context_channel_fx)
+        # make sure it tasted awful
+        assert not async_callable_fx.was_called
+
+    @pytest.mark.parametrize("sep", [',', ';', '|', '-'])
+    @pytest.mark.parametrize("platform_str, platform",
+                             [("pc", Platforms.PC), ("ps", Platforms.PS), ("ps4", Platforms.PS),
+                              ("playstation", Platforms.PS), ("playstation4", Platforms.PS),
+                              ("Playstation 4", Platforms.PS), ("xb", Platforms.XB),
+                              ("xb1", Platforms.XB), ("xbox", Platforms.XB),
+                              ("xbox one", Platforms.XB)
+                              ]
+                             )
+    async def test_manual_rsig_handler(self,
+                                       async_callable_fx: AsyncCallableMock,
+                                       context_channel_fx: Context,
+                                       monkeypatch,
+                                       sep: chr,
+                                       platform_str: str,
+                                       platform: Platforms
+                                       ):
+        """
+        Tests with multiple cases, that the parser recognized the case details
+        and creates an appropriate rescue
+        """
+
+        # again, tracking stuff needs to be set up
+        monkeypatch.setattr(context_channel_fx, 'reply', async_callable_fx)
+        rat_board = context_channel_fx.bot.board
+
+        # give us a message
+        monkeypatch.setattr(context_channel_fx, '_words_eol',
+                            [f"ratsignal H{sep} {platform_str}{sep} O2 OK"]
+                            )
+
+        # and a nickname
+        monkeypatch.setattr(context_channel_fx._user, '_nickname', "Absolver")
+        # throw it into the magic black box
+        await RatMama.handle_selfissued_ratsignal(context_channel_fx)
+        # and remember the result
+        case = rat_board.find_by_name("Absolver")
+
+        # assert all details are as expected
+        assert case is not None
+        assert case.platform == platform
+        assert case.system.casefold() == "h"
+        assert case.irc_nickname.casefold() == "absolver"
+        assert not case.code_red
+
+        # who needs flash when they can have cleanse?
+        rat_board.clear_board()
+
+        # now lets get a bit more mean with the message
+        monkeypatch.setattr(context_channel_fx, '_words_eol',
```

Comment:
[Suggestion]  Can we use pytests' parametrize function to test more strings here?  I like the idea of pulling as much data as possible, but some TDD would be great here.

Reply:
👍